agent/agent.go

```diff
@@ -17,18 +20,27 @@ import (
 	"go.uber.org/zap"
 )
 
+var byteOrder = binary.BigEndian
+
 var connectionHelper agentConnectionHelper = newAgentConnectionHelper()
 
-type Agent struct {
+type Agent interface {
+	Listen(ipString string, port int) error
+	Connect(Agent) error
+	Disconnect(Agent) error
+	SendMessage(size int) error
+	Stop() error
+	NumPeers() int
+}
+
+type GossipAgent struct {
 	Host   host.Host
 	Logger *zap.Logger
-	Config *AgentConfig
+	Config *GossipConfig
```

Comment:
Could we address variable changes in another PR later on? It is hard to follow the changes, much of the diff lines are changing variable names.

Reply:
sure thing 👍